engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -391,58 +398,43 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
-        WorkflowSourceFiles(workflowAux.wdlSource.toRawString, workflowAux.jsonInputs.toRawString, workflowAux.workflowOptions.toRawString)
-      )
-    } yield workflowDescriptor
+    } yield ExecutionAndAux(workflowExecutionResult, workflowAux)
 
-    runTransaction(action)
+    runTransaction(action) flatMap futureDescriptorFromExecutionAndAux
   }
 
   override def getWorkflow(workflowId: WorkflowId)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
     val action = for {
       workflowExecutionResult <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
       workflowAux <- dataAccess.workflowExecutionAuxesByWorkflowExecutionUuid(workflowExecutionResult.workflowExecutionUuid).result.head
-      workflowDescriptor = WorkflowDescriptor(
-        workflowId,
-        WorkflowSourceFiles(workflowAux.wdlSource.toRawString, workflowAux.jsonInputs.toRawString, workflowAux.workflowOptions.toRawString)
-      )
-    } yield workflowDescriptor
+    } yield ExecutionAndAux(workflowExecutionResult, workflowAux)
 
-    runTransaction(action)
+    runTransaction(action) flatMap futureDescriptorFromExecutionAndAux
   }
 
   override def getWorkflowsByState(states: Traversable[WorkflowState])
                                   (implicit ec: ExecutionContext): Future[Traversable[WorkflowDescriptor]] = {
     val action = for {
       workflowExecutionResults <- dataAccess.workflowExecutionsByStatuses(states.map(_.toString)).result
+      workflowAuxSeq <- DBIO.sequence(workflowExecutionResults.map( workflowExecutionResult =>
+        dataAccess.workflowExecutionAuxesByWorkflowExecutionUuid(workflowExecutionResult.workflowExecutionUuid).result.head
+      ))
+      executionAndAux <- DBIO.sequence((workflowExecutionResults zip workflowAuxSeq) map { case (exec, aux) => DBIO.successful(ExecutionAndAux(exec, aux)) })
+    } yield executionAndAux
 
-      workflowDescriptors <- DBIO.sequence(
-        workflowExecutionResults map { workflowExecutionResult =>
-          val workflowExecutionAuxResult = dataAccess.workflowExecutionAuxesByWorkflowExecutionUuid(
-            workflowExecutionResult.workflowExecutionUuid).result.head
-
-          workflowExecutionAuxResult map { workflowExecutionAux =>
-            WorkflowDescriptor(
-              WorkflowId.fromString(workflowExecutionResult.workflowExecutionUuid),
-              WorkflowSourceFiles(
-                workflowExecutionAux.wdlSource.toRawString,
-                workflowExecutionAux.jsonInputs.toRawString,
-                workflowExecutionAux.workflowOptions.toRawString
-              )
-            )
-          }
-        }
-      )
-
-    } yield workflowDescriptors
-
-    runTransaction(action)
+    for {
+      executionAndAuxes <- runTransaction(action)
+      workflowDescSeq <- Future.sequence(executionAndAuxes.map(exAux => futureDescriptorFromExecutionAndAux(exAux)))
+    } yield workflowDescSeq
```

Comment:
I wish I understood Slick better, but is there a way to do this in a better way?